operator/pkg/ciliumendpointslice/endpointslice.go

```diff
@@ -59,30 +59,61 @@ func (c *Controller) initializeQueue() {
 		logfields.WorkQueueBurstLimit:  c.rateLimit.current.Burst,
 		logfields.WorkQueueSyncBackOff: defaultSyncBackOff,
 	}).Info("CES controller workqueue configuration")
-	c.queue = workqueue.NewRateLimitingQueueWithConfig(
-		workqueue.NewItemExponentialFailureRateLimiter(defaultSyncBackOff, maxSyncBackOff),
+	c.rateLimiter = workqueue.NewItemExponentialFailureRateLimiter(defaultSyncBackOff, maxSyncBackOff)
+	c.fastQueue = workqueue.NewRateLimitingQueueWithConfig(
+		c.rateLimiter,
+		workqueue.RateLimitingQueueConfig{Name: "cilium_endpoint_slice"})
+	c.standardQueue = workqueue.NewRateLimitingQueueWithConfig(
+		c.rateLimiter,
 		workqueue.RateLimitingQueueConfig{Name: "cilium_endpoint_slice"})
 }
 
 func (c *Controller) onEndpointUpdate(cep *cilium_api_v2.CiliumEndpoint) {
 	if cep.Status.Networking == nil || cep.Status.Identity == nil || cep.GetName() == "" || cep.Namespace == "" {
 		return
 	}
+	c.logger.Infof("on CEP update %s, %s", cep.Name, cep.Namespace)
 	touchedCESs := c.manager.UpdateCEPMapping(k8s.ConvertCEPToCoreCEP(cep), cep.Namespace)
 	c.enqueueCESReconciliation(touchedCESs)
 }
 
 func (c *Controller) onEndpointDelete(cep *cilium_api_v2.CiliumEndpoint) {
+	c.logger.Infof("on CEP delete %s, %s", cep.Name, cep.Namespace)
 	touchedCES := c.manager.RemoveCEPMapping(k8s.ConvertCEPToCoreCEP(cep), cep.Namespace)
 	c.enqueueCESReconciliation([]CESName{touchedCES})
 }
 
 func (c *Controller) onSliceUpdate(ces *capi_v2a1.CiliumEndpointSlice) {
-	c.enqueueCESReconciliation([]CESName{NewCESName(ces.Name)})
+	c.logger.Infof("on CES update %s, %s", ces.Name, ces.Namespace)
+	c.enqueueCESReconciliation([]CESName{NewCESNameNamespace(ces.Name, ces.Namespace)})
 }
 
 func (c *Controller) onSliceDelete(ces *capi_v2a1.CiliumEndpointSlice) {
-	c.enqueueCESReconciliation([]CESName{NewCESName(ces.Name)})
+	c.logger.Infof("on CES delete %s, %s", ces.Name, ces.Namespace)
+	c.enqueueCESReconciliation([]CESName{NewCESNameNamespace(ces.Name, ces.Namespace)})
+}
+
+func (c *Controller) addToQueue(ces CESName) {
+	_, ok := c.priorityNamespaces[ces.Namespace]
+
+	if ok {
+		time.AfterFunc(c.defaultCESSyncTime, func() {
+			c.logger.Infof("CES %s, %s added to the fast queue after %d", ces.Name, ces.Namespace, c.defaultCESSyncTime)
+			c.fastQueue.Add(ces)
```

Comment:
we could potentially check the `if exists` condition here instead of having two time.AfterFunc